diff -urN xmms-1.2.6.org/Input/vorbis/Makefile.am xmms-1.2.6/Input/vorbis/Makefile.am
--- xmms-1.2.6.org/Input/vorbis/Makefile.am	Thu Jan 17 23:25:37 2002
+++ xmms-1.2.6/Input/vorbis/Makefile.am	Thu Jan 17 23:26:28 2002
@@ -8,7 +8,7 @@
 
 INCLUDES = @GTK_CFLAGS@ @OGG_CFLAGS@ @VORBIS_CFLAGS@ -I$(top_builddir)/intl
 libvorbis_la_LDFLAGS = @PLUGIN_LDFLAGS@
-libvorbis_la_LIBADD = @GTK_LIBS@ @PTHREAD_LIBS@ @VORBIS_LIBS@ @OGG_LIBS@
+libvorbis_la_LIBADD = @GTK_LIBS@ @PTHREAD_LIBS@ @VORBIS_LIBS@ @VORBISFILE_LIBS@ @OGG_LIBS@
 
 libvorbis_la_SOURCES = \
 vorbis.c vorbis.h \
diff -urN xmms-1.2.6.org/configure.in xmms-1.2.6/configure.in
--- xmms-1.2.6.org/configure.in	Thu Jan 17 23:25:35 2002
+++ xmms-1.2.6/configure.in	Thu Jan 17 23:25:57 2002
@@ -37,6 +37,7 @@
 
 dnl Various compiler setup.
 AC_PROG_CC
+AM_PROG_AS
 AC_ISC_POSIX
 dnl AC_PROG_CXX
 AC_C_BIGENDIAN
diff -urN xmms-1.2.6.org/xmms/Makefile.am xmms-1.2.6/xmms/Makefile.am
--- xmms-1.2.6.org/xmms/Makefile.am	Thu Jan 17 23:25:36 2002
+++ xmms-1.2.6/xmms/Makefile.am	Thu Jan 17 23:25:57 2002
@@ -5,7 +5,7 @@
 bin_PROGRAMS = xmms
 
 xmms_LDFLAGS = -export-dynamic
-xmms_LDADD += @GTK_LIBS@ @PTHREAD_LIBS@ @SM_LIBS@ @VM_LIBS@ \
+xmms_LDADD = @GTK_LIBS@ @PTHREAD_LIBS@ @SM_LIBS@ @VM_LIBS@ \
 @POSIX_LIBS@ $(top_builddir)/libxmms/libxmms.la @INTLLIBS@
 
 INCLUDES = @GTK_CFLAGS@ @XMMS_DEFINES@ @ARCH_DEFINES@ -I$(top_builddir)/intl
