diff -urN xmms-1.2.2.orig/wmxmms/getopt.c xmms-1.2.2/wmxmms/getopt.c
--- xmms-1.2.2.orig/wmxmms/getopt.c	Thu Jul 29 23:03:12 1999
+++ xmms-1.2.2/wmxmms/getopt.c	Sun Aug 20 21:39:30 2000
@@ -89,7 +89,7 @@
    When compiling libc, the _ macro is predefined.  */
 #ifdef HAVE_LIBINTL_H
 #include <libintl.h>
-#define _(msgid)	gettext (msgid)
+#define _(msgid)	dgettext (PACKAGE, msgid)
 #else
 #define _(msgid)	(msgid)
 #endif
diff -urN xmms-1.2.2.orig/xmms/about.c xmms-1.2.2/xmms/about.c
--- xmms-1.2.2.orig/xmms/about.c	Mon Jun 12 15:27:14 2000
+++ xmms-1.2.2/xmms/about.c	Sun Aug 20 21:40:09 2000
@@ -131,14 +131,14 @@
 		gchar *temp[2];
 		guint row;
 		
-		temp[0] = gettext(credit_text[i++]);
-		temp[1] = gettext(credit_text[i++]);
+		temp[0] = _(credit_text[i++]);
+		temp[1] = _(credit_text[i++]);
 		row = gtk_clist_append(GTK_CLIST(clist), temp);
 		gtk_clist_set_selectable(GTK_CLIST(clist), row, FALSE);
 		temp[0] = "";
 		while (credit_text[i])
 		{
-			temp[1] = gettext(credit_text[i++]);
+			temp[1] = _(credit_text[i++]);
 			row = gtk_clist_append(GTK_CLIST(clist), temp);
 			gtk_clist_set_selectable(GTK_CLIST(clist), row, FALSE);
 		}
diff -urN xmms-1.2.2.orig/xmms/getopt.c xmms-1.2.2/xmms/getopt.c
--- xmms-1.2.2.orig/xmms/getopt.c	Thu Jul 29 23:03:12 1999
+++ xmms-1.2.2/xmms/getopt.c	Sun Aug 20 21:40:48 2000
@@ -89,7 +89,7 @@
    When compiling libc, the _ macro is predefined.  */
 #ifdef HAVE_LIBINTL_H
 #include <libintl.h>
-#define _(msgid)	gettext (msgid)
+#define _(msgid)	dgettext (PACKAGE, msgid)
 #else
 #define _(msgid)	(msgid)
 #endif
diff -urN xmms-1.2.2.orig/xmms/i18n.h xmms-1.2.2/xmms/i18n.h
--- xmms-1.2.2.orig/xmms/i18n.h	Sun Feb  6 13:54:09 2000
+++ xmms-1.2.2/xmms/i18n.h	Sun Aug 20 21:40:30 2000
@@ -5,7 +5,7 @@
 #ifdef ENABLE_NLS
 #  include <locale.h>
 #    include <libintl.h>
-#    define _(String) gettext (String)
+#    define _(String) dgettext (PACKAGE, String)
 #    ifdef gettext_noop
 #        define N_(String) gettext_noop (String)
 #    else
