--- xmms-1.2.10/Output/alsa/audio.c.orig	2005-03-22 11:01:44.000000000 +0100
+++ xmms-1.2.10/Output/alsa/audio.c	2005-03-22 11:02:24.000000000 +0100
@@ -499,7 +499,7 @@
 #define MONO_ADJUST(type, type2, endian)					\
 do {										\
 	type *ptr = data;							\
-	for (i = 0; i < length; i += 4)						\
+	for (i = 0; i < length; i += 2)						\
 	{									\
 		*ptr = type2##_TO_##endian(type2##_FROM_## endian(*ptr) *	\
 					   vol / 100);				\
@@ -530,7 +530,7 @@
 #define MONO_ADJUST8(type)			\
 do {						\
 	type *ptr = data;			\
-	for (i = 0; i < length; i += 4)		\
+	for (i = 0; i < length; i++)		\
 	{					\
 		*ptr = *ptr * vol / 100;	\
 		ptr++;				\
