diff -Nuard xmms-1.2.8.orig/xmms/about.c xmms-1.2.8/xmms/about.c
--- xmms-1.2.8.orig/xmms/about.c	2003-09-04 15:48:34.000000000 +0200
+++ xmms-1.2.8/xmms/about.c	2003-09-08 01:18:57.000000000 +0200
@@ -75,6 +75,8 @@
 	N_("Chris Wilson"),
 	N_("Dave Yearke"),
 	N_("Stephan K. Zitz"),
+	/* for languages that can't display 'l slash' (³) replace it with "l" */
+	N_("Micha³ Kochanowicz"),
 	NULL,
  N_("Default skin:"),
  	N_("Leonard \"Blayde\" Tan"),
--- xmms-1.2.11/xmms/input.c.orig	2005-05-15 02:01:21.000000000 +0200
+++ xmms-1.2.11/xmms/input.c	2007-11-17 22:33:23.903143854 +0100
@@ -224,6 +224,72 @@
 	pthread_mutex_unlock(&vis_mutex);
 }
 
+void input_unplayable_window_destroyed(GtkObject *object, gpointer user_data)
+{
+	*((GtkWidget **)user_data) = NULL;
+}
+
+void input_dont_show_warrning(GtkObject *object, gpointer user_data)
+{
+	*((gboolean *)user_data) = !gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(object));
+}
+
+void input_show_unplayable_file_warrning(gchar * filename)
+{
+	static GtkWidget *unplayables_window = NULL;
+	static GtkWidget *list;
+
+	GList *l = NULL;
+
+	if(!unplayables_window) {
+		GtkWidget *label, *button, *scrolled;
+		GtkWidget *check;
+		GtkWidget *vbox;
+
+		unplayables_window = gtk_window_new(GTK_WINDOW_DIALOG);
+		gtk_window_set_default_size(GTK_WINDOW(unplayables_window), 300, 400);
+		gtk_window_set_title(GTK_WINDOW(unplayables_window), _("Unplayable files"));
+		gtk_window_set_policy(GTK_WINDOW(unplayables_window), FALSE, TRUE, FALSE);
+		gtk_container_set_border_width(GTK_CONTAINER(unplayables_window), 10);
+		gtk_signal_connect(GTK_OBJECT(unplayables_window), "destroy",
+				   GTK_SIGNAL_FUNC(input_unplayable_window_destroyed), &unplayables_window);
+		gtk_widget_realize(unplayables_window);
+		
+		vbox = gtk_vbox_new(FALSE, 4);
+		label = gtk_label_new("Following files could not be played. They are either not accessible or require dedicated input plugin (xmms-input-*-.rpm).");
+		scrolled = gtk_scrolled_window_new(NULL, NULL);
+		list = gtk_list_new();
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolled), list);
+		check = gtk_check_button_new_with_label("don't show this warrning any more");
+		button = gtk_button_new_with_label("close");
+
+		gtk_signal_connect_object(GTK_OBJECT(button), "clicked",
+			GTK_SIGNAL_FUNC(gtk_widget_destroy),
+			GTK_OBJECT(unplayables_window));
+
+		gtk_signal_connect(GTK_OBJECT(check), "clicked",
+			GTK_SIGNAL_FUNC(input_dont_show_warrning), &cfg.warn_about_unplayables);
+
+		gtk_box_pack_start(GTK_BOX(vbox), label, FALSE, FALSE, 0);
+		gtk_box_pack_start(GTK_BOX(vbox), scrolled, TRUE, TRUE, 0);
+		gtk_box_pack_start(GTK_BOX(vbox), check, FALSE, FALSE, 0);
+		gtk_box_pack_start(GTK_BOX(vbox), button, FALSE, FALSE, 0);
+		
+		gtk_container_add(GTK_CONTAINER(unplayables_window), vbox);
+	}
+	
+	l = g_list_append(l, gtk_list_item_new_with_label(filename));
+	gtk_list_append_items(GTK_LIST(list), l);
+	
+	gtk_widget_show_all(unplayables_window);
+}
+
+void input_file_not_playable(gchar * filename)
+{
+	if(cfg.warn_about_unplayables)
+		input_show_unplayable_file_warrning(filename);
+}
+
 gboolean input_check_file(gchar * filename)
 {
 	GList *node;
@@ -238,6 +304,7 @@
 			return TRUE;
 		node = node->next;
 	}
+	input_file_not_playable(filename);
 	return FALSE;
 }
 
@@ -274,6 +341,7 @@
 		}
 		node = node->next;
 	}
+	input_file_not_playable(filename);
 	set_current_input_plugin(NULL);
 }
 
diff -Nuard xmms-1.2.8.orig/xmms/main.c xmms-1.2.8/xmms/main.c
--- xmms-1.2.8.orig/xmms/main.c	2003-09-02 15:01:40.000000000 +0200
+++ xmms-1.2.8/xmms/main.c	2003-09-08 01:17:19.000000000 +0200
@@ -346,6 +346,7 @@
 	cfg.random_skin_on_play = FALSE;
 	cfg.mainwin_use_xfont = FALSE;
 	cfg.use_pl_metadata = TRUE;
+	cfg.warn_about_unplayables = TRUE;
 
 	cfg.playlist_x = 295;
 	cfg.playlist_y = 20;
@@ -397,6 +398,7 @@
 		xmms_cfg_read_boolean(cfgfile, "xmms", "no_playlist_advance", &cfg.no_playlist_advance);
 		xmms_cfg_read_boolean(cfgfile, "xmms", "sort_jump_to_file", &cfg.sort_jump_to_file);
 		xmms_cfg_read_boolean(cfgfile, "xmms", "use_pl_metadata", &cfg.use_pl_metadata);
+		xmms_cfg_read_boolean(cfgfile, "xmms", "warn_about_unplayables", &cfg.warn_about_unplayables);
 		xmms_cfg_read_boolean(cfgfile, "xmms", "smooth_title_scroll", &cfg.smooth_title_scroll);
 		xmms_cfg_read_boolean(cfgfile, "xmms", "use_backslash_as_dir_delimiter", &cfg.use_backslash_as_dir_delimiter);
 		xmms_cfg_read_int(cfgfile, "xmms", "player_x", &cfg.player_x);
@@ -556,6 +558,7 @@
 	xmms_cfg_write_boolean(cfgfile, "xmms", "save_window_positions", cfg.save_window_position);
 	xmms_cfg_write_boolean(cfgfile, "xmms", "dim_titlebar", cfg.dim_titlebar);
 	xmms_cfg_write_boolean(cfgfile, "xmms", "use_pl_metadata", cfg.use_pl_metadata);
+	xmms_cfg_write_boolean(cfgfile, "xmms", "warn_about_unplayables", cfg.warn_about_unplayables);
 	xmms_cfg_write_boolean(cfgfile, "xmms", "get_info_on_load", cfg.get_info_on_load);
 	xmms_cfg_write_boolean(cfgfile, "xmms", "get_info_on_demand", cfg.get_info_on_demand);
 	xmms_cfg_write_boolean(cfgfile, "xmms", "eq_doublesize_linked", cfg.eq_doublesize_linked);
diff -Nuard xmms-1.2.8.orig/xmms/main.h xmms-1.2.8/xmms/main.h
--- xmms-1.2.8.orig/xmms/main.h	2002-10-06 18:35:27.000000000 +0200
+++ xmms-1.2.8/xmms/main.h	2003-09-08 01:17:36.000000000 +0200
@@ -42,6 +42,7 @@
 	gboolean get_info_on_demand, eq_doublesize_linked;
 	gboolean sort_jump_to_file, use_eplugins, always_on_top, sticky;
 	gboolean no_playlist_advance, smooth_title_scroll, use_pl_metadata;
+	gboolean warn_about_unplayables;
 	gboolean use_backslash_as_dir_delimiter;
 	gboolean random_skin_on_play, use_fontsets;
 	gboolean mainwin_use_xfont;
diff -Nuard xmms-1.2.8.orig/xmms/prefswin.c xmms-1.2.8/xmms/prefswin.c
--- xmms-1.2.8.orig/xmms/prefswin.c	2003-07-16 15:17:47.000000000 +0200
+++ xmms-1.2.8/xmms/prefswin.c	2003-09-08 01:18:14.000000000 +0200
@@ -978,6 +978,7 @@
 	gtk_box_pack_start(GTK_BOX(options_gi_box), options_giol, FALSE, FALSE, 0);
 	gtk_table_attach_defaults(GTK_TABLE(options_table), options_gi_box, 0, 1, 0, 1);
 	prefswin_option_new_with_label_to_table(&cfg.allow_multiple_instances, _("Allow multiple instances"), GTK_TABLE(options_table), 1, 0);
+	prefswin_option_new_with_label_to_table(&cfg.warn_about_unplayables, _("Warn about not playable files"), GTK_TABLE(options_table), 0, 10);
 
 	prefswin_option_new_with_label_to_table(&cfg.convert_twenty, _("Convert %20 to space"), GTK_TABLE(options_table), 0, 1);
 	opt = prefswin_option_new_with_label_to_table(&cfg.always_show_cb, _("Always show clutterbar"), GTK_TABLE(options_table), 1, 1);
