diff -urN xmms-1.2.6.org/Input/vorbis/Makefile.am xmms-1.2.6/Input/vorbis/Makefile.am
--- xmms-1.2.6.org/Input/vorbis/Makefile.am	Thu Jan 17 23:25:37 2002
+++ xmms-1.2.6/Input/vorbis/Makefile.am	Thu Jan 17 23:26:28 2002
@@ -8,7 +8,7 @@
 
 INCLUDES = @GTK_CFLAGS@ @OGG_CFLAGS@ @VORBIS_CFLAGS@ -I$(top_builddir)/intl
 libvorbis_la_LDFLAGS = @PLUGIN_LDFLAGS@
-libvorbis_la_LIBADD = @GTK_LIBS@ @PTHREAD_LIBS@ @VORBIS_LIBS@ @OGG_LIBS@
+libvorbis_la_LIBADD = @GTK_LIBS@ @PTHREAD_LIBS@ @VORBIS_LIBS@ @VORBISFILE_LIBS@ @OGG_LIBS@
 
 libvorbis_la_SOURCES = \
 vorbis.c vorbis.h \
diff -urN xmms-1.2.6.org/configure.in xmms-1.2.6/configure.in
--- xmms-1.2.6.org/configure.in	Thu Jan 17 23:25:35 2002
+++ xmms-1.2.6/configure.in	Thu Jan 17 23:25:57 2002
@@ -37,6 +37,7 @@
 
 dnl Various compiler setup.
 AC_PROG_CC
+AM_PROG_AS
 AC_ISC_POSIX
 dnl AC_PROG_CXX
 AC_C_BIGENDIAN
diff -urN xmms-1.2.6.org/xmms/Makefile.am xmms-1.2.6/xmms/Makefile.am
--- xmms-1.2.6.org/xmms/Makefile.am	Thu Jan 17 23:25:36 2002
+++ xmms-1.2.6/xmms/Makefile.am	Thu Jan 17 23:25:57 2002
@@ -5,7 +5,7 @@
 bin_PROGRAMS = xmms
 
 xmms_LDFLAGS = -export-dynamic
-xmms_LDADD += @GTK_LIBS@ @PTHREAD_LIBS@ @SM_LIBS@ @VM_LIBS@ \
+xmms_LDADD = @GTK_LIBS@ @PTHREAD_LIBS@ @SM_LIBS@ @VM_LIBS@ \
 @POSIX_LIBS@ $(top_builddir)/libxmms/libxmms.la @INTLLIBS@
 
 INCLUDES = @GTK_CFLAGS@ @XMMS_DEFINES@ @ARCH_DEFINES@ -I$(top_builddir)/intl
--- xmms-1.2.7/acinclude.m4.orig	Mon Jun 10 20:14:41 2002
+++ xmms-1.2.7/acinclude.m4	Mon Jun 10 20:23:05 2002
@@ -49,6 +49,8 @@
 	    AC_PATH_PROG(GNOME_CONFIG,gnome-config,no)
 	    if test "$GNOME_CONFIG" = "no"; then
 	      no_gnome_config="yes"
+	      AM_CONDITIONAL(HAVE_ORBIT,false)
+	      AM_CONDITIONAL(HAVE_GNORBA,false)
 	    else
 	      AC_MSG_CHECKING(if $GNOME_CONFIG works)
 	      if $GNOME_CONFIG --libs-only-l gnome >/dev/null 2>&1; then
@@ -65,6 +66,8 @@
 	      else
 	        AC_MSG_RESULT(no)
 	        no_gnome_config="yes"
+		AM_CONDITIONAL(HAVE_ORBIT,false)
+		AM_CONDITIONAL(HAVE_GNORBA,false)
               fi
             fi
 
@@ -93,6 +95,9 @@
  	        fi
 	      fi
             fi
+	else
+	  AM_CONDITIONAL(HAVE_ORBIT,false)
+	  AM_CONDITIONAL(HAVE_GNORBA,false)
 	fi
 
 	if test -n "$3"; then
