diff -Nur xmms-1.2.3.org/Makefile.am xmms-1.2.3/Makefile.am
--- xmms-1.2.3.org/Makefile.am	Mon Jul 31 23:32:15 2000
+++ xmms-1.2.3/Makefile.am	Sun Oct 29 16:25:24 2000
@@ -1,4 +1,4 @@
-SUBDIRS = intl libxmms xmms Output Input Effect General Visualization wmxmms gnomexmms po
+SUBDIRS = vorbis-plugins intl libxmms xmms Output Input Effect General Visualization wmxmms gnomexmms po
 
 bin_SCRIPTS = xmms-config
 
diff -Nur xmms-1.2.3.org/configure.in xmms-1.2.3/configure.in
--- xmms-1.2.3.org/configure.in	Sun Oct 29 16:19:47 2000
+++ xmms-1.2.3/configure.in	Sun Oct 29 16:31:20 2000
@@ -244,6 +244,9 @@
 AC_CHECK_HEADERS(sched.h)
 AC_CHECK_HEADERS(sys/sched.h)
 
+AC_CHECK_LIB(vorbis, vorbis_info_init, LIBS="$LIBS -lvorbis", AC_MSG_ERROR([You need libvorbis library to compile xmms-vorbis plugin!!!]))
+AC_CHECK_LIB(vorbisfile, ov_open, LIBS="$LIBS -lvorbisfile", AC_MSG_ERROR([You need libvorbisfile library to compile xmms-vorbis plugin!!!]))
+
 AC_CHECK_FUNC(sched_setscheduler,,[AC_CHECK_LIB(posix4,sched_setscheduler)])
 if test x$ac_cv_func_sched_setscheduler = x && test x$ac_cv_lib_sched_setscheduler = xno 
 then :
@@ -332,6 +335,8 @@
 Makefile
 xmms.spec
 xmms-config
+vorbis-plugins/Makefile
+vorbis-plugins/xmms/Makefile
 xmms/Makefile
 xmms/defskin/Makefile
 Output/Makefile
diff -Nur xmms-1.2.3.org/vorbis-plugins/Makefile.am xmms-1.2.3/vorbis-plugins/Makefile.am
--- xmms-1.2.3.org/vorbis-plugins/Makefile.am	Thu Jan  1 01:00:00 1970
+++ xmms-1.2.3/vorbis-plugins/Makefile.am	Sun Oct 29 16:24:56 2000
@@ -0,0 +1,2 @@
+SUBDIRS = xmms
+
diff -Nur xmms-1.2.3.org/vorbis-plugins/xmms/Makefile.am xmms-1.2.3/vorbis-plugins/xmms/Makefile.am
--- xmms-1.2.3.org/vorbis-plugins/xmms/Makefile.am	Thu Jan  1 01:00:00 1970
+++ xmms-1.2.3/vorbis-plugins/xmms/Makefile.am	Sun Oct 29 17:18:07 2000
@@ -0,0 +1,11 @@
+lib_LTLIBRARIES = libvorbis.la
+
+libdir = $(plugindir)/$(INPUT_PLUGIN_DIR)
+
+libvorbis_la_LDFLAGS = -module -avoid-version
+
+libvorbis_la_LIBADD = @GTK_LIBS@
+
+libvorbis_la_SOURCES = vorbis.c
+
+INCLUDES = @GTK_CFLAGS@ -I$(top_builddir)/intl
diff -Nur xmms-1.2.3.org/vorbis-plugins/xmms/vorbis.c xmms-1.2.3/vorbis-plugins/xmms/vorbis.c
--- xmms-1.2.3.org/vorbis-plugins/xmms/vorbis.c	Sun Jul 30 15:09:10 2000
+++ xmms-1.2.3/vorbis-plugins/xmms/vorbis.c	Sun Oct 29 17:05:44 2000
@@ -15,10 +15,9 @@
 #include <string.h>
 #include <pthread.h>
 #include <glib.h>
-#include "xmms/plugin.h"
-#include "xmms/util.h"
-#include "vorbis/codec.h"
-#include "vorbis/vorbisfile.h"
+#include <xmms/plugin.h>
+#include <vorbis/codec.h>
+#include <vorbis/vorbisfile.h>
 
 #define BLOCK_SIZE 4096
 
