--- xmms-1.0.1/xmms/pluginenum.c.aarrgh	Tue Feb 22 17:51:17 2000
+++ xmms-1.0.1/xmms/pluginenum.c	Tue Feb 22 17:50:07 2000
@@ -71,6 +71,7 @@
 	InputPlugin *ip;
 	EffectPlugin *ep;
 	gint dirsel = 0;
+	gint found = 0;
 
 	if (cfg.disabled_iplugins)
 	{
@@ -122,11 +123,27 @@
 	while (node)
 	{
 		op = (OutputPlugin *) node->data;
-		if (!strcmp(cfg.outputplugin, op->filename))
+		if (!strcmp(cfg.outputplugin, op->filename)) {
 			op_data->current_output_plugin = op;
+			found = 1;
+		}
 		if (op->init)
 			op->init();
 		node = node->next;
+	}
+	/* HACK - If they have configured a plugin that doesn't
+	 * exist, try and find the next best thing. */
+	if (found == 0) {
+		node = op_data->output_list;
+		while (node)
+		{
+			op = (OutputPlugin *) node->data;
+			if (!strcmp(basename(cfg.outputplugin), basename(op->filename)))
+			  op_data->current_output_plugin = op;
+			if (op->init)
+			  op->init();
+			node = node->next;
+		}
 	}
 
 	node = ep_data->effect_list;
