diff -Nuard xmms-1.2.8.warn/configure.in xmms-1.2.8/configure.in
--- xmms-1.2.8.warn/configure.in	2003-09-04 23:00:18.000000000 +0200
+++ xmms-1.2.8/configure.in	2003-09-08 03:15:27.000000000 +0200
@@ -64,8 +64,9 @@
 
 AC_CHECK_HEADER(pthread.h,,AC_MSG_ERROR([*** POSIX thread support not installed - please install first ***]))
 
-AM_PATH_GLIB(1.2.2,,AC_MSG_ERROR([*** GLIB >= 1.2.2 not installed - please install first ***]))
-AM_PATH_GTK(1.2.2,,AC_MSG_ERROR([*** GTK+ >= 1.2.2 not installed - please install first ***]),gthread)
+AM_PATH_GLIB_2_0(2.2.0,,AC_MSG_ERROR([*** GLIB >= 2.2.0 not installed - please install first ***]))
+AM_PATH_GTK_2_0(2.2.0,,AC_MSG_ERROR([*** GTK+ >= 2.2.0 not installed - please install first ***]),gthread)
+
 AC_PATH_PROG(XMMS_PATH,xmms,no)
 
 LIBS_save=$LIBS
diff -Nuard xmms-1.2.8.warn/General/ir/about.c xmms-1.2.8/General/ir/about.c
--- xmms-1.2.8.warn/General/ir/about.c	2000-02-16 22:05:50.000000000 +0100
+++ xmms-1.2.8/General/ir/about.c	2003-09-08 03:15:27.000000000 +0200
@@ -22,7 +22,7 @@
 
 	if (ir_about_win)
 		return;
-	ir_about_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	ir_about_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(ir_about_win), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &ir_about_win);
 	gtk_window_set_title(GTK_WINDOW(ir_about_win), _("About"));
 	gtk_window_set_policy(GTK_WINDOW(ir_about_win), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/General/ir/configure.c xmms-1.2.8/General/ir/configure.c
--- xmms-1.2.8.warn/General/ir/configure.c	2002-09-09 21:47:55.000000000 +0200
+++ xmms-1.2.8/General/ir/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -141,7 +141,7 @@
 
 		keepConfGoing = TRUE;
 		irbutton_to_edit = data;
-		irconf_controlwin = gtk_window_new(GTK_WINDOW_DIALOG);
+		irconf_controlwin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_signal_connect(GTK_OBJECT(irconf_controlwin), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &irconf_controlwin);
 		tmp=g_strdup_printf(_("`%s' Button Setup"),_(data));
 		gtk_window_set_title(GTK_WINDOW(irconf_controlwin), tmp);
@@ -250,7 +250,7 @@
 	if (!irconf_mainwin)
 	{
 
-		irconf_mainwin = gtk_window_new(GTK_WINDOW_DIALOG);
+		irconf_mainwin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_signal_connect(GTK_OBJECT(irconf_mainwin), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &irconf_mainwin);
 		gtk_window_set_title(GTK_WINDOW(irconf_mainwin), _("XMMS IRman Configuration"));
 		gtk_window_set_policy(GTK_WINDOW(irconf_mainwin), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/General/joystick/configure.c xmms-1.2.8/General/joystick/configure.c
--- xmms-1.2.8.warn/General/joystick/configure.c	2002-01-07 15:08:01.000000000 +0100
+++ xmms-1.2.8/General/joystick/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -89,7 +89,7 @@
 
 	if (!joyconf_mainwin)
 	{
-		joyconf_mainwin = gtk_window_new(GTK_WINDOW_DIALOG);
+		joyconf_mainwin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_signal_connect(GTK_OBJECT(joyconf_mainwin), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &joyconf_mainwin);
 		gtk_window_set_title(GTK_WINDOW(joyconf_mainwin), _("XMMS Joystick Configuration"));
 		gtk_window_set_policy(GTK_WINDOW(joyconf_mainwin), FALSE, FALSE, FALSE);
@@ -182,7 +182,7 @@
 		/* -------------------------------------------------- */
 		frame = gtk_frame_new(_("Directionals:"));
 		gtk_box_pack_start(GTK_BOX(box), frame, FALSE, FALSE, 0);
-		dir_pack = gtk_packer_new();
+		dir_pack = gtk_hbox_new(FALSE, 10);
 		gtk_container_set_border_width(GTK_CONTAINER(dir_pack), 5);
 		gtk_container_add(GTK_CONTAINER(frame), dir_pack);
 		hist_val[0] = joy_cfg.up;
@@ -194,7 +194,7 @@
 		{
 			blist = gtk_option_menu_new();
 			gtk_widget_set_usize(blist, 120, -1);
-			gtk_packer_add(GTK_PACKER(dir_pack), blist, pack_pos[i], GTK_ANCHOR_CENTER, 0, 0, 5, 5, 0, 0);
+			gtk_box_pack_start(GTK_BOX(dir_pack), blist, TRUE, TRUE, 0);
 			gtk_option_menu_remove_menu(GTK_OPTION_MENU(blist));
 			gtk_option_menu_set_menu(GTK_OPTION_MENU(blist), joy_menus[i]);
 			gtk_option_menu_set_history(GTK_OPTION_MENU(blist), hist_val[i]);
@@ -207,7 +207,7 @@
 		/* -------------------------------------------------- */
 		frame = gtk_frame_new(_("Directionals (alternate):"));
 		gtk_box_pack_start(GTK_BOX(box), frame, FALSE, FALSE, 0);
-		dir_pack = gtk_packer_new();
+		dir_pack = gtk_hbox_new(FALSE, 10);
 		gtk_container_set_border_width(GTK_CONTAINER(dir_pack), 5);
 		gtk_container_add(GTK_CONTAINER(frame), dir_pack);
 		hist_val[0] = joy_cfg.alt_up;
@@ -219,7 +219,7 @@
 		{
 			blist = gtk_option_menu_new();
 			gtk_widget_set_usize(blist, 120, -1);
-			gtk_packer_add(GTK_PACKER(dir_pack), blist, pack_pos[i], GTK_ANCHOR_CENTER, 0, 0, 5, 5, 0, 0);
+			gtk_box_pack_start(GTK_BOX(dir_pack), blist, TRUE, TRUE, 0);
 			gtk_option_menu_remove_menu(GTK_OPTION_MENU(blist));
 			gtk_option_menu_set_menu(GTK_OPTION_MENU(blist), joy_menus[4 + i]);
 			gtk_option_menu_set_history(GTK_OPTION_MENU(blist), hist_val[i]);
diff -Nuard xmms-1.2.8.warn/General/song_change/song_change.c xmms-1.2.8/General/song_change/song_change.c
--- xmms-1.2.8.warn/General/song_change/song_change.c	2003-07-14 15:24:28.000000000 +0200
+++ xmms-1.2.8/General/song_change/song_change.c	2003-09-08 03:15:27.000000000 +0200
@@ -114,7 +114,7 @@
 	GtkWidget *warn_win, *warn_vbox, *warn_desc;
 	GtkWidget *warn_bbox, *warn_yes, *warn_no;
 
-	warn_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	warn_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(warn_win), _("Warning"));
 	gtk_window_set_transient_for(GTK_WINDOW(warn_win),
 				     GTK_WINDOW(configure_win));
@@ -201,7 +201,7 @@
 
 	read_config();
 
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(configure_win), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &configure_win);
 	gtk_window_set_title(GTK_WINDOW(configure_win), _("Song Change Configuration"));
 	
diff -Nuard xmms-1.2.8.warn/Input/cdaudio/cddb.c xmms-1.2.8/Input/cdaudio/cddb.c
--- xmms-1.2.8.warn/Input/cdaudio/cddb.c	2003-07-03 16:08:36.000000000 +0200
+++ xmms-1.2.8/Input/cdaudio/cddb.c	2003-09-08 03:15:27.000000000 +0200
@@ -838,7 +838,7 @@
 	if (debug_window)
 		return;
 	
-	debug_window = gtk_window_new(GTK_WINDOW_DIALOG);
+	debug_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(debug_window), "destroy",
 			   GTK_SIGNAL_FUNC(gtk_widget_destroyed), &debug_window);
 	gtk_window_set_title(GTK_WINDOW(debug_window), "CDDB networkdebug");
diff -Nuard xmms-1.2.8.warn/Input/cdaudio/configure.c xmms-1.2.8/Input/cdaudio/configure.c
--- xmms-1.2.8.warn/Input/cdaudio/configure.c	2003-09-04 23:00:18.000000000 +0200
+++ xmms-1.2.8/Input/cdaudio/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -328,7 +328,7 @@
 					  directory);
 	}
 			
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_transient_for(GTK_WINDOW(window),
 				     GTK_WINDOW(cdda_configure_win));
 	gtk_container_set_border_width(GTK_CONTAINER(window), 10);
@@ -497,7 +497,7 @@
 	if (cdda_configure_win)
 		return;
 	
-	cdda_configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	cdda_configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(cdda_configure_win), "destroy",
 			   GTK_SIGNAL_FUNC(gtk_widget_destroyed),
 			   &cdda_configure_win);
diff -Nuard xmms-1.2.8.warn/Input/mikmod/plugin.c xmms-1.2.8/Input/mikmod/plugin.c
--- xmms-1.2.8.warn/Input/mikmod/plugin.c	2001-07-23 14:26:27.000000000 +0200
+++ xmms-1.2.8/Input/mikmod/plugin.c	2003-09-08 03:15:27.000000000 +0200
@@ -435,7 +435,7 @@
 
 	if (!mikmod_conf_window)
 	{
-		mikmod_conf_window = gtk_window_new(GTK_WINDOW_DIALOG);
+		mikmod_conf_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_object_set_data(GTK_OBJECT(mikmod_conf_window), "mikmod_conf_window", mikmod_conf_window);
 		gtk_window_set_title(GTK_WINDOW(mikmod_conf_window), _("MikMod Configuration"));
 		gtk_window_set_policy(GTK_WINDOW(mikmod_conf_window), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/Input/mpg123/configure.c xmms-1.2.8/Input/mpg123/configure.c
--- xmms-1.2.8.warn/Input/mpg123/configure.c	2002-10-06 18:35:23.000000000 +0200
+++ xmms-1.2.8/Input/mpg123/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -239,7 +239,7 @@
 		gdk_window_raise(mpg123_configurewin->window);
 		return;
 	}
-	mpg123_configurewin = gtk_window_new(GTK_WINDOW_DIALOG);
+	mpg123_configurewin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(mpg123_configurewin), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &mpg123_configurewin);
 	gtk_signal_connect(GTK_OBJECT(mpg123_configurewin), "destroy", GTK_SIGNAL_FUNC(configure_destroy), &mpg123_configurewin);
 	gtk_window_set_title(GTK_WINDOW(mpg123_configurewin), _("MPG123 Configuration"));
diff -Nuard xmms-1.2.8.warn/Input/mpg123/fileinfo.c xmms-1.2.8/Input/mpg123/fileinfo.c
--- xmms-1.2.8.warn/Input/mpg123/fileinfo.c	2003-07-14 15:24:28.000000000 +0200
+++ xmms-1.2.8/Input/mpg123/fileinfo.c	2003-09-08 03:15:27.000000000 +0200
@@ -248,7 +248,7 @@
 		GtkWidget *label, *filename_hbox;
 		GtkWidget *bbox, *save, *remove_id3, *cancel;
 		
-		window = gtk_window_new(GTK_WINDOW_DIALOG);
+		window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_window_set_policy(GTK_WINDOW(window), FALSE, FALSE, FALSE);
 		gtk_signal_connect(GTK_OBJECT(window), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &window);
 		gtk_container_set_border_width(GTK_CONTAINER(window), 10);
diff -Nuard xmms-1.2.8.warn/Input/vorbis/configure.c xmms-1.2.8/Input/vorbis/configure.c
--- xmms-1.2.8.warn/Input/vorbis/configure.c	2002-10-06 18:35:26.000000000 +0200
+++ xmms-1.2.8/Input/vorbis/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -220,7 +220,7 @@
 		return;
 	}
 
-	vorbis_configurewin = gtk_window_new(GTK_WINDOW_DIALOG);
+	vorbis_configurewin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(vorbis_configurewin), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &vorbis_configurewin);
 	gtk_signal_connect(GTK_OBJECT(vorbis_configurewin), "destroy", GTK_SIGNAL_FUNC(configure_destroy), &vorbis_configurewin);
 	gtk_window_set_title(GTK_WINDOW(vorbis_configurewin), _("Ogg Vorbis Configuration"));
diff -Nuard xmms-1.2.8.warn/Input/vorbis/fileinfo.c xmms-1.2.8/Input/vorbis/fileinfo.c
--- xmms-1.2.8.warn/Input/vorbis/fileinfo.c	2002-07-23 18:28:16.000000000 +0200
+++ xmms-1.2.8/Input/vorbis/fileinfo.c	2003-09-08 03:15:27.000000000 +0200
@@ -454,7 +454,7 @@
 		GtkWidget *table, *bbox, *cancel_button;
 		GtkWidget *save_button, *remove_button;
 
-		window = gtk_window_new(GTK_WINDOW_DIALOG);
+		window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_window_set_policy(GTK_WINDOW(window), FALSE, FALSE, FALSE);
 		gtk_signal_connect(GTK_OBJECT(window), "destroy", 
 			GTK_SIGNAL_FUNC(gtk_widget_destroyed), &window);
diff -Nuard xmms-1.2.8.warn/libxmms/dirbrowser.c xmms-1.2.8/libxmms/dirbrowser.c
--- xmms-1.2.8.warn/libxmms/dirbrowser.c	2002-11-18 16:06:54.000000000 +0100
+++ xmms-1.2.8/libxmms/dirbrowser.c	2003-09-08 03:15:27.000000000 +0200
@@ -276,7 +276,7 @@
 	GtkCTree *ctree;
 	struct dirnode *dirnode;
 
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_default_size(GTK_WINDOW(window), 300, 400);
 	gtk_window_set_title(GTK_WINDOW(window), title);
 	gtk_container_border_width(GTK_CONTAINER(window), 10);
diff -Nuard xmms-1.2.8.warn/Output/alsa/configure.c xmms-1.2.8/Output/alsa/configure.c
--- xmms-1.2.8.warn/Output/alsa/configure.c	2003-07-21 14:53:18.000000000 +0200
+++ xmms-1.2.8/Output/alsa/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -160,7 +160,7 @@
 		return;
 	}
 
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(configure_win), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &configure_win);
 	gtk_window_set_title(GTK_WINDOW(configure_win), _("ALSA Driver configuration"));
 	gtk_window_set_policy(GTK_WINDOW(configure_win), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/Output/disk_writer/disk_writer.c xmms-1.2.8/Output/disk_writer/disk_writer.c
--- xmms-1.2.8.warn/Output/disk_writer/disk_writer.c	2002-01-08 15:09:07.000000000 +0100
+++ xmms-1.2.8/Output/disk_writer/disk_writer.c	2003-09-08 03:15:27.000000000 +0200
@@ -338,7 +338,7 @@
 {
 	if (!configure_win)
 	{
-		configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+		configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 
 		gtk_signal_connect(GTK_OBJECT(configure_win), "destroy", GTK_SIGNAL_FUNC(configure_destroy), NULL);
 		gtk_signal_connect(GTK_OBJECT(configure_win), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &configure_win);
diff -Nuard xmms-1.2.8.warn/Output/esd/configure.c xmms-1.2.8/Output/esd/configure.c
--- xmms-1.2.8.warn/Output/esd/configure.c	2001-12-03 15:42:17.000000000 +0100
+++ xmms-1.2.8/Output/esd/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -87,7 +87,7 @@
 		return;
 	}
 
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(configure_win), "destroy",
 			   GTK_SIGNAL_FUNC(gtk_widget_destroyed),
 			   &configure_win);
diff -Nuard xmms-1.2.8.warn/Output/OSS/configure.c xmms-1.2.8/Output/OSS/configure.c
--- xmms-1.2.8.warn/Output/OSS/configure.c	2001-08-07 14:46:07.000000000 +0200
+++ xmms-1.2.8/Output/OSS/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -175,7 +175,7 @@
 		return;
 	}
 
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(configure_win), "destroy", GTK_SIGNAL_FUNC(gtk_widget_destroyed), &configure_win);
 	gtk_window_set_title(GTK_WINDOW(configure_win), _("OSS Driver configuration"));
 	gtk_window_set_policy(GTK_WINDOW(configure_win), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/Output/sun/configure.c xmms-1.2.8/Output/sun/configure.c
--- xmms-1.2.8.warn/Output/sun/configure.c	2003-01-07 15:50:09.000000000 +0100
+++ xmms-1.2.8/Output/sun/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -529,7 +529,7 @@
 		gdk_window_raise(configure_win->window);
 		return;
 	}
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(configure_win), "destroy",
 			   GTK_SIGNAL_FUNC(configure_win_destroy), NULL);
 
diff -Nuard xmms-1.2.8.warn/Visualization/blur_scope/blur_scope.c xmms-1.2.8/Visualization/blur_scope/blur_scope.c
--- xmms-1.2.8.warn/Visualization/blur_scope/blur_scope.c	2000-11-24 20:11:52.000000000 +0100
+++ xmms-1.2.8/Visualization/blur_scope/blur_scope.c	2003-09-08 03:15:27.000000000 +0200
@@ -153,7 +153,7 @@
 		return;
 	bscope_read_config();
 
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(window),_("Blur scope"));
 	gtk_window_set_policy(GTK_WINDOW(window), FALSE, FALSE, FALSE);
 	gtk_widget_realize(window);
diff -Nuard xmms-1.2.8.warn/Visualization/blur_scope/config.c xmms-1.2.8/Visualization/blur_scope/config.c
--- xmms-1.2.8.warn/Visualization/blur_scope/config.c	2002-09-09 21:47:56.000000000 +0200
+++ xmms-1.2.8/Visualization/blur_scope/config.c	2003-09-08 03:15:27.000000000 +0200
@@ -63,7 +63,7 @@
 	color[1]=((gdouble)((bscope_cfg.color %0x10000)/0x100))/256;
 	color[2]=((gdouble)(bscope_cfg.color %0x100))/256;
 	
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_container_set_border_width(GTK_CONTAINER(configure_win), 10);
 	gtk_window_set_title(GTK_WINDOW(configure_win), _("Color Entry"));
 	gtk_window_set_policy(GTK_WINDOW(configure_win), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/Visualization/opengl_spectrum/configure.c xmms-1.2.8/Visualization/opengl_spectrum/configure.c
--- xmms-1.2.8.warn/Visualization/opengl_spectrum/configure.c	2001-06-01 15:07:49.000000000 +0200
+++ xmms-1.2.8/Visualization/opengl_spectrum/configure.c	2003-09-08 03:15:27.000000000 +0200
@@ -38,7 +38,7 @@
 
 	oglspectrum_read_config();
 
-	configure_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_container_set_border_width(GTK_CONTAINER(configure_win), 10);
 	gtk_window_set_title(GTK_WINDOW(configure_win), _("OpenGL Spectrum configuration"));
 	gtk_window_set_policy(GTK_WINDOW(configure_win), FALSE, FALSE, FALSE);
diff -Nuard xmms-1.2.8.warn/Visualization/sanalyzer/spectrum.c xmms-1.2.8/Visualization/sanalyzer/spectrum.c
--- xmms-1.2.8.warn/Visualization/sanalyzer/spectrum.c	2000-07-31 23:41:05.000000000 +0200
+++ xmms-1.2.8/Visualization/sanalyzer/spectrum.c	2003-09-08 03:15:27.000000000 +0200
@@ -82,7 +82,7 @@
 	int i;
 	if(window)
 		return;
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(window),_("Spectrum analyzer"));
 	gtk_window_set_policy(GTK_WINDOW(window), FALSE, FALSE, FALSE);
 	gtk_widget_realize(window);
diff -Nuard xmms-1.2.8.warn/wmxmms/wmxmms.c xmms-1.2.8/wmxmms/wmxmms.c
--- xmms-1.2.8.warn/wmxmms/wmxmms.c	2002-10-28 16:24:23.000000000 +0100
+++ xmms-1.2.8/wmxmms/wmxmms.c	2003-09-08 03:15:27.000000000 +0200
@@ -102,10 +102,6 @@
 	{"text/plain", 0, 1}
 };
 
-/* Does anyone know a better way? */
-
-extern Window gdk_leader_window;
-
 void action_play(void)
 {
 	xmms_remote_play(xmms_session);
@@ -584,7 +580,7 @@
 		gtk_tooltips_set_delay(tooltips, 1000);
 	}
 
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	if (has_geometry)
 		gtk_widget_set_uposition(window, win_x, win_y);
 	gtk_widget_set_usize(window, 64, 64);
@@ -619,7 +615,7 @@
 	hints.flags = StateHint;
 	XSetWMHints(GDK_DISPLAY(), GDK_WINDOW_XWINDOW(window->window), &hints);
 
-	icon_win = gtk_window_new(GTK_WINDOW_DIALOG);
+	icon_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_widget_set_app_paintable(icon_win, TRUE);
 	gtk_widget_set_uposition(icon_win, 0, 0);
 	gtk_widget_set_usize(icon_win, 64, 64);
@@ -689,7 +685,7 @@
 	pixmap = gdk_pixmap_create_from_xpm_d(window->window,
 					      &mask, NULL, xmms_dock_master_xpm);
 
-	leader = gdk_window_foreign_new(gdk_leader_window);
+	leader = gdk_window_foreign_new((GdkNativeWindow)gdk_x11_get_default_root_xwindow());
 	gdk_window_set_icon(leader, icon_win->window, NULL, NULL);
 	gdk_window_reparent(icon_win->window, leader, 0, 0);
 
diff -Nuard xmms-1.2.8.warn/xmms/about.c xmms-1.2.8/xmms/about.c
--- xmms-1.2.8.warn/xmms/about.c	2003-09-08 03:14:38.000000000 +0200
+++ xmms-1.2.8/xmms/about.c	2003-09-08 03:15:27.000000000 +0200
@@ -229,7 +229,7 @@
 	if (about_window)
 		return;
 	
-	about_window = gtk_window_new(GTK_WINDOW_DIALOG);
+	about_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(about_window), _("About XMMS"));
 	gtk_window_set_policy(GTK_WINDOW(about_window), FALSE, TRUE, FALSE);
 	gtk_container_set_border_width(GTK_CONTAINER(about_window), 10);
diff -Nuard xmms-1.2.8.warn/xmms/equalizer.c xmms-1.2.8/xmms/equalizer.c
--- xmms-1.2.8.warn/xmms/equalizer.c	2003-04-02 16:45:05.000000000 +0200
+++ xmms-1.2.8/xmms/equalizer.c	2003-09-08 03:15:27.000000000 +0200
@@ -659,7 +659,7 @@
 
 static void equalizerwin_create_gtk(void)
 {
-	equalizerwin = gtk_window_new(GTK_WINDOW_DIALOG);
+	equalizerwin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	dock_add_window(dock_window_list, equalizerwin);
 	gtk_widget_set_app_paintable(equalizerwin, TRUE);
 	gtk_window_set_policy(GTK_WINDOW(equalizerwin), FALSE, FALSE, TRUE);
@@ -1201,7 +1201,7 @@
 	char *preset_text[1];
 	GList *node;
 
-	*window = gtk_window_new(GTK_WINDOW_DIALOG);
+	*window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_signal_connect(GTK_OBJECT(*window), "destroy",
 			   GTK_SIGNAL_FUNC(gtk_widget_destroyed), window);
 	gtk_window_set_transient_for(GTK_WINDOW(*window), GTK_WINDOW(equalizerwin));
@@ -1554,7 +1554,7 @@
 	GtkWidget *vbox, *hbox, *instructions;
 	GtkWidget *ok, *cancel, *apply;
 
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(window), _("Configure Equalizer"));
 	gtk_window_set_policy(GTK_WINDOW(window), FALSE, FALSE, FALSE);
 	gtk_window_set_transient_for(GTK_WINDOW(window), GTK_WINDOW(mainwin));
diff -Nuard xmms-1.2.8.warn/xmms/input.c xmms-1.2.8/xmms/input.c
--- xmms-1.2.8.warn/xmms/input.c	2003-09-08 03:14:38.000000000 +0200
+++ xmms-1.2.8/xmms/input.c	2003-09-08 03:15:45.000000000 +0200
@@ -246,7 +246,7 @@
 		GtkWidget *check;
 		GtkWidget *vbox;
 
-		unplayables_window = gtk_window_new(GTK_WINDOW_DIALOG);
+		unplayables_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 		gtk_window_set_default_size(GTK_WINDOW(unplayables_window), 300, 400);
 		gtk_window_set_title(GTK_WINDOW(unplayables_window), _("Unplayable files"));
 		gtk_window_set_policy(GTK_WINDOW(unplayables_window), FALSE, TRUE, FALSE);
@@ -455,7 +455,7 @@
 
 	char *title, *iplugin;
 		
-	window = gtk_window_new(GTK_WINDOW_DIALOG);
+	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_policy(GTK_WINDOW(window), FALSE, FALSE, FALSE);
 	title = g_strdup_printf(_("File Info - %s"), g_basename(filename));
 	gtk_window_set_title(GTK_WINDOW(window), title);
diff -Nuard xmms-1.2.8.warn/xmms/main.c xmms-1.2.8/xmms/main.c
--- xmms-1.2.8.warn/xmms/main.c	2003-09-08 03:14:38.000000000 +0200
+++ xmms-1.2.8/xmms/main.c	2003-09-08 03:15:50.000000000 +0200
@@ -1417,7 +1417,7 @@
 	if (!get_input_playing())
 		return;
 
-	mainwin_jtt = gtk_window_new(GTK_WINDOW_DIALOG);
+	mainwin_jtt = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(mainwin_jtt), _("Jump to time"));
 	gtk_window_set_position(GTK_WINDOW(mainwin_jtt), GTK_WIN_POS_CENTER);
 	gtk_window_set_policy(GTK_WINDOW(mainwin_jtt), FALSE, FALSE, FALSE);
@@ -1745,7 +1745,7 @@
 		return;
 	}
 
-	mainwin_jtf = gtk_window_new(GTK_WINDOW_DIALOG);
+	mainwin_jtf = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(mainwin_jtf), _("Jump to file"));
 	gtk_window_set_position(GTK_WINDOW(mainwin_jtf), GTK_WIN_POS_CENTER);
 	gtk_window_set_transient_for(GTK_WINDOW(mainwin_jtf), GTK_WINDOW(mainwin));
diff -Nuard xmms-1.2.8.warn/xmms/playlistwin.c xmms-1.2.8/xmms/playlistwin.c
--- xmms-1.2.8.warn/xmms/playlistwin.c	2003-04-04 19:42:51.000000000 +0200
+++ xmms-1.2.8/xmms/playlistwin.c	2003-09-08 03:15:27.000000000 +0200
@@ -1884,7 +1884,7 @@
 
 static void playlistwin_create_gtk(void)
 {
-	playlistwin = gtk_window_new(GTK_WINDOW_DIALOG);
+	playlistwin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	dock_add_window(dock_window_list, playlistwin);
 	gtk_widget_set_app_paintable(playlistwin, TRUE);
 	if (cfg.show_wm_decorations)
diff -Nuard xmms-1.2.8.warn/xmms/prefswin.c xmms-1.2.8/xmms/prefswin.c
--- xmms-1.2.8.warn/xmms/prefswin.c	2003-09-08 03:14:38.000000000 +0200
+++ xmms-1.2.8/xmms/prefswin.c	2003-09-08 03:15:27.000000000 +0200
@@ -715,7 +715,7 @@
 
 	char *titles[1];
 
-	prefswin = gtk_window_new(GTK_WINDOW_DIALOG);
+	prefswin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(prefswin), _("Preferences"));
 	gtk_window_set_policy(GTK_WINDOW(prefswin), FALSE, FALSE, FALSE);
 	gtk_window_set_transient_for(GTK_WINDOW(prefswin), GTK_WINDOW(mainwin));
diff -Nuard xmms-1.2.8.warn/xmms/skinwin.c xmms-1.2.8/xmms/skinwin.c
--- xmms-1.2.8.warn/xmms/skinwin.c	2001-06-17 01:05:47.000000000 +0200
+++ xmms-1.2.8/xmms/skinwin.c	2003-09-08 03:15:27.000000000 +0200
@@ -46,7 +46,7 @@
 	char *titles[1];
 	GtkWidget *vbox, *hbox, *main_hbox, *separator, *scrolled_win, *checkbox;
 
-	skinwin = gtk_window_new(GTK_WINDOW_DIALOG);
+	skinwin = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(skinwin), _("Skin selector"));
 	gtk_window_set_transient_for(GTK_WINDOW(skinwin), GTK_WINDOW(mainwin));
 	gtk_signal_connect(GTK_OBJECT(skinwin), "delete_event", GTK_SIGNAL_FUNC(skinwin_delete_event), NULL);
diff -Nuard xmms-1.2.8.warn/xmms/util.c xmms-1.2.8/xmms/util.c
--- xmms-1.2.8.warn/xmms/util.c	2003-09-02 15:01:40.000000000 +0200
+++ xmms-1.2.8/xmms/util.c	2003-09-08 03:15:27.000000000 +0200
@@ -553,7 +553,7 @@
 {
 	GtkWidget *win, *vbox, *bbox, *ok, *enqueue, *cancel, *combo;
 	
-	win = gtk_window_new(GTK_WINDOW_DIALOG);
+	win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 	gtk_window_set_title(GTK_WINDOW(win), caption);
 	gtk_window_set_position(GTK_WINDOW(win), GTK_WIN_POS_MOUSE);
 	gtk_window_set_default_size(GTK_WINDOW(win), 400, -1);
@@ -969,14 +969,14 @@
 void util_dump_menu_rc(void)
 {
 	char *filename = g_strconcat(g_get_home_dir(), "/.xmms/menurc", NULL);
-	gtk_item_factory_dump_rc(filename, NULL, FALSE);
+	gtk_accel_map_save(filename);
 	g_free(filename);
 }
 
 void util_read_menu_rc(void)
 {
 	char *filename = g_strconcat(g_get_home_dir(), "/.xmms/menurc", NULL);
-	gtk_item_factory_parse_rc(filename);
+	gtk_accel_map_load(filename);
 	g_free(filename);
 }
 
