--- xmms-0.9.1/Output/disk_writer/disk_writer.c.divzero	Wed Aug 18 17:56:52 1999
+++ xmms-0.9.1/Output/disk_writer/disk_writer.c	Wed Aug 18 17:57:10 1999
@@ -253,7 +253,7 @@
 
 static gint disk_get_written_time(void)
 {
-	return (gint) (((gfloat) written * 1000) / (gfloat) (header.byte_p_sec));
+	return header.byte_p_sec ? (gint) (((gfloat) written * 1000) / (gfloat) (header.byte_p_sec)) : 0;
 }
 
 static gint disk_get_output_time(void)
