--- xmms-0.9.1/Output/OSS/audio.c.foo	Wed Sep 22 19:21:24 1999
+++ xmms-0.9.1/Output/OSS/audio.c	Wed Sep 22 19:26:02 1999
@@ -476,12 +476,13 @@
 		device_name = g_strdup_printf("/dev/dsp%d", oss_cfg.audio_device);
 	else
 		device_name = g_strdup("/dev/dsp");
-	fd = open(device_name, O_WRONLY);
+	fd = open(device_name, O_WRONLY|O_NONBLOCK);
 	if (fd == -1)
 	{
 		g_free(buffer);
 		return 0;
 	}
+	fcntl(fd,F_SETFL,fcntl(fd,F_GETFL) &~ O_NONBLOCK);
 	oss_set_audio_params();
 	if (!realtime)
 		pthread_create(&buffer_thread, NULL, oss_loop, NULL);
