--- xmms-1.2.10/libxmms/xmmsctrl.h.orig	2003-06-11 20:44:17.000000000 +0200
+++ xmms-1.2.10/libxmms/xmmsctrl.h	2004-05-08 11:29:28.259765480 +0200
@@ -85,7 +85,7 @@
 
 	
 #ifdef __cplusplus
-};
+}
 #endif
 
 /* Deprecated APIs */
--- xmms-1.2.10/General/ir/ir.c.orig	2001-03-05 14:17:44.000000000 +0100
+++ xmms-1.2.10/General/ir/ir.c	2005-03-17 18:49:17.000000000 +0100
@@ -16,10 +16,10 @@
 #include "ir.h"
 
 /* Important stuff to know */
-static gboolean keepGoing = FALSE;
+gboolean keepGoing = FALSE;
 
 /* The thread handle */
-static pthread_t irapp_thread;
+pthread_t irapp_thread;
 
 /* Declarations for calls that we need to mention in the plugin struct */
 static void init(void);
