diff -urN xmms-1.2.2.orig/Output/OSS/audio.c xmms-1.2.2/Output/OSS/audio.c
--- xmms-1.2.2.orig/Output/OSS/audio.c	Wed Jul 12 13:53:05 2000
+++ xmms-1.2.2/Output/OSS/audio.c	Wed Jul 12 13:52:32 2000
@@ -592,7 +592,7 @@
 		else
 			device_name = g_strdup("/dev/dsp");
 	}
-	fd = open(device_name, O_WRONLY);
+	fd = open(device_name, O_WRONLY|O_NONBLOCK);
 	if (fd == -1)
 	{
 		g_warning("oss_open(): Failed to open audio device (%s): %s",
@@ -600,6 +600,7 @@
 		g_free(device_name);
 		return 0;
 	}
+	fcntl(fd,F_SETFL,fcntl(fd,F_GETFL) &~ O_NONBLOCK);
 
 	oss_setup_format(fmt,rate,nch);
 	
diff -urN xmms-1.2.2.orig/xmms/main.c xmms-1.2.2/xmms/main.c
--- xmms-1.2.2.orig/xmms/main.c	Tue Jul 11 18:11:14 2000
+++ xmms-1.2.2/xmms/main.c	Wed Jul 12 13:51:01 2000
@@ -310,7 +310,7 @@
 
 	memset(&cfg, 0, sizeof (Config));
 #if defined(HAVE_SYS_SOUNDCARD_H) || defined(HAVE_MACHINE_SOUNDCARD_H)
-	cfg.outputplugin = g_strdup_printf("%s/%s/libOSS.so", PLUGIN_DIR, plugin_dir_list[0]);
+	cfg.outputplugin = g_strdup_printf("%s/%s/libesdout.so", PLUGIN_DIR, plugin_dir_list[0]);
 #else
 	cfg.outputplugin = g_strdup("");
 #endif
